sample/ConsoleDemo/Program.cs

```diff
@@ -1,13 +1,63 @@
 ﻿using Serilog;
 using Serilog.Sinks.SystemConsole.Themes;
 using System;
+using System.Linq;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace ConsoleDemo
 {
     public class Program
     {
-        public static void Main()
+        public static void Main(string[] args)
+        {
+            if (args != null && args.Length == 1)
+            {
+                switch (args[0])
+                {
+                    case "--sync-root-default":
+                        SystemConsoleSyncTest(syncRootForLogger1: null, syncRootForLogger2: null);
+                        return;
+                    case "--sync-root-separate":
+                        SystemConsoleSyncTest(syncRootForLogger1: new object(), syncRootForLogger2: new object());
+                        return;
+                    case "--sync-root-same":
+                    {
+                        var sameSyncRoot = new object();
+                        SystemConsoleSyncTest(syncRootForLogger1: sameSyncRoot, syncRootForLogger2: sameSyncRoot);
+                        return;
+                    }
+                }
+            }
+            
+            AnsiConsoleThemeTest();
+        }
+
+        static void SystemConsoleSyncTest(object syncRootForLogger1, object syncRootForLogger2)
+        {
+            var logger1 = new LoggerConfiguration()
+                .MinimumLevel.Verbose()
+                .Enrich.WithProperty("Logger", "logger1")
+                .WriteTo.Console(theme: SystemConsoleTheme.Literate, syncRoot: syncRootForLogger1)
+                .CreateLogger();
+
+            var logger2 = new LoggerConfiguration()
+                .MinimumLevel.Verbose()
+                .Enrich.WithProperty("Logger", "logger2")
+                .WriteTo.Console(theme: SystemConsoleTheme.Literate, syncRoot: syncRootForLogger2)
+                .CreateLogger();
+
+            var options = new ParallelOptions { MaxDegreeOfParallelism = 8 };
+            System.Threading.Tasks.Parallel.For(0, 1000, options, (i, loopState) =>
+            {
+                var logger = (i % 2 == 0) ? logger1 : logger2;
+                logger.Information("Event {Iteration} generated by {ThreadId}", i, Thread.CurrentThread.ManagedThreadId);
+            });
+
+            Log.CloseAndFlush();
+        }
+
+        static void AnsiConsoleThemeTest()
```

Comment:
The demo's good, but we should probably keep the example super-simple and revert this file before merging.

Reply:
You're right, i'll revert the sample changes.

Do you think it deserves moving the "synchronized writes" sample into a new `csproj` in the sample folder, or deleting is best? It's not particularly easy to demo the issue, that's all.

Reply:
Ah, sorry, missed this - including it as a second sample with a new name sounds good 👍

Reply:
added sample in ad280ad